design-proposals/klip-41-ksqldb-.net-client.md

```diff
@@ -0,0 +1,72 @@
+# KLIP 41 - ksqlDB .NET Client
+
+**Author**: Alex Basiuk (@alex-basiuk) | 
+**Release Target**: TBD | 
+**Status**: _In Discussion_ | 
+**Discussion**: TBD
+
+**tl;dr:** _Create a ksqlDB client for .NET which is on par with the existing Java client.
+The .NET community will benefit from the this project. The goal is to provide first class developer experience and 
+simplify writing event streaming applications in .NET._
+
+## Motivation and background
+
+Basically, it's similar to [KLIP 15](./klip-15-new-api-and-client.md).
+It should lower the barrier for .NET developers and increase adoption of ksqlDB by the community.
+The library should offer the same feature set as the Java library and should follow common .NET conventions and patterns.  
+
+## What is in scope
+
+* A new client implemented in C#.
+* Initially, the client will be targeted at .NET Core 3 and .NET Framework 5. Support of older versions will be added later.
```

Comment:
I'm not versed in .NET, but one thing I'm wondering is how support for multiple versions will look over time. Any ideas about how to manage that workflow?

Reply:
@MichaelDrogalis .NET 5 and .NET Core 3 are very close in terms of APIs. I don't think .NET 5 has any breaking changes in the libraries required by this project. It won't be hard to support both [until end of support of .NET Core 3.1](https://dotnet.microsoft.com/platform/support/policy/dotnet-core). Basically, you can have one project which [targets multiple framework versions](https://docs.microsoft.com/en-us/dotnet/standard/frameworks).
Support of older versions is probably more involved.

Reply:
Cool, that makes sense. Mostly just asking since Im unfamiliar with the .NET stack.